controllers/reconcilers/grafana/deployment_reconciler.go

```diff
@@ -211,6 +220,41 @@ func getContainers(cr *v1beta1.Grafana, scheme *runtime.Scheme, vars *v1beta1.Op
 	return containers
 }
 
+// getGrafanaContainerSecurityContext provides default securityContext for grafana container
+func getGrafanaContainerSecurityContext() *v1.SecurityContext {
+	capability := &v1.Capabilities{
+		Drop: []v1.Capability{
+			"ALL",
+		},
+	}
+	return &v1.SecurityContext{
+		AllowPrivilegeEscalation: model.BoolPtr(false),
+		ReadOnlyRootFilesystem:   model.BoolPtr(true),
+		Privileged:               model.BoolPtr(false),
+		RunAsNonRoot:             model.BoolPtr(true),
+		RunAsUser:                model.IntPtr(10001),
```

Comment:
We need to check if we are running on OCP before setting these default settings.

Reply:
Hmm, there's that or we can set validation to false, which would ignore unknown fields on the kubernetes custom resource so we wouldn't need to do env-specific logic, but that's also not an ideal situation, it can lead to misconfigurations quite easily. (so doing logic for this check is probably a must anwyay)

Reply:
How does it work in OCP nowadays? Does it mutate the `RunAsUser` field of individual pods? And when doing so, does it simply ignore the existing value?

And does it really need to be set? I mean wouldn't OCP consider an image with a `USER <non-root>` instruction set sufficient? - OCP doesn't work on m1, so can't check myself :)

Reply:
@HubertStefanski yeah I think we want to keep validation, I reached out in the operator-sdk-dev channel in k8s and asked for the easiest workaround and got a reply: https://kubernetes.slack.com/archives/C017UU45SHL/p1671050125336449
OpenTelemetry collector operator seems to be doing something similar: https://github.com/open-telemetry/opentelemetry-operator/blob/b7081575f8742cddbf82b7e59c33b405882ba8af/pkg/autodetect/main.go#L74

So sadly OCP don't mutate the `RunAsUser` filed (unless this have changed the last 2 years).
Instead it just say that you can set RunAsUser due to SCC or something like that.

It needs to be set if we want to set `RunAsNonRoot` or some of the other you can't run as root rule. I don't remember which one it is.

Reply:
@NissesSenap I think `RunAsNonRoot` only makes a pod (or a container?) fail in case container tries to start with `uid` 0, it does not specifically demand for `RunAsUser`. With it enabled, either of those two cases would satisfy the restriction:
- If an image has a `USER` instruction with a non-zero `uid` (or refers to a non-`root` username);
- If a container has `RunAsUser` (in its own spec or derived from pod's `securityContext`) with a non-zero `uid`.

```
~  docker run --rm -it --entrypoint sh grafana/grafana -c id
uid=472(grafana) gid=0(root) groups=0(root)
```

So, a custom `uid` is already chosen. From what I remember, the issue with OCP was that it required to use a specific range for uid which was different for each of the namespaces we had.

Reply:
Yes exactly so that is why we can't define any RunAsUser on OCP. I will write some logic to check for OCP and not set it.